packages/react/src/lib/factories.ts

```diff
@@ -10,6 +10,7 @@ import {
   ShorthandRenderer,
 } from '../types'
 import { mergeStyles } from './mergeThemes'
+import { UIComponentProps } from '.'
```

Comment:
Ouch, circular reference

```suggestion
import { UIComponentProps } from './commonPropInterfaces'
```